ghc.nix

```diff
@@ -27,10 +27,11 @@ in
 , withNuma ? (pkgsFor nixpkgs system).stdenv.isLinux
 , withDtrace ? (pkgsFor nixpkgs system).stdenv.isLinux
 , withGrind ? !((pkgsFor nixpkgs system).valgrind.meta.broken or false)
+, withSystemLibffi ? false
```

Comment:
This one lgtm, though not necessary for the wasm backend itself; the `wasi-sdk` setup hook takes care of the libffi business.

Reply:
I'll be honest, this was an unrelated commit that I had in my tree.